### ISSUE TITLE
Bug: default: in wrong place?

### ISSUE THREAD

**Issue by mpogue2** [mpogue2/SquareDesk#404]
```
    switch (songFilenameFormat) {
        default:  // WARNING: THIS IS ALMOST CERTAINLY WRONG
        case SongFilenameLabelDashName :
            return label_first_matches;
        case SongFilenameNameDashLabel :
            return filename_first_matches;
        case SongFilenameBestGuess :
            return best_guess_matches;
    }
```
@danlyke I added the comment (based on warnings I'm now seeing with the latest warning defaults in QtCreator), but I'm not really sure what you intended here.  I bet it's not what's written, though!  Could you take a look at this?

BTW, the warning message is "default label in switch which covers all enumeration values".  :-)

**Comment by mpogue2** [comment]
If you don't have time to dive into this, please take a look at the code below -- is this what was intended?

```
    switch (songFilenameFormat) {
        case SongFilenameNameDashLabel :
            return filename_first_matches;

        case SongFilenameBestGuess :
            return best_guess_matches;

        case SongFilenameLabelDashName :
        default:
            return label_first_matches;
    }
```

**Comment by danlyke** [comment]
I think that default placement was probably intentional: If we, for
some reason, have an errant songFileFormat, use label_first_matches.

But an extra return in there, and a logged warning, wouldn't be a bad
thing.

**Comment by danlyke** [comment]
That looks like a fair rework.

**Comment by mpogue2** [comment]
I can't get rid of the warning messages on Mac OS X (you are welcome to try!).  But, it's basically fixed by c630189440721746da5dc8b330e3822421d4ae69 .
